sql/executor.go

```diff
@@ -236,9 +234,9 @@ func (e Executor) execStmt(stmt parser.Statement, params parameters, planMaker *
 	// No transaction. Run the command as a retryable block in an
 	// auto-transaction.
 	err := e.db.Txn(func(txn *client.Txn) error {
-		planMaker.txn = txn
+		planMaker.assignTxn(txn, timestamp)
```

Comment:
well the clock continues to run - consider the case of an insert statement using transaction_timestamp: this implementation allow drift in the face of retries